### PR TITLE
simply syncd start logic on different platforms (requires one image)

### PR DESCRIPTION
**Pull request by lguohan**

### CODE REVIEW

**Comment by jleveque** [comment]
WIth this PR, your should also be able to delete the etc/syncd.d directory from this repo.

File: debian/syncd.init [hunk 1]
```diff
@@ -34,68 +51,47 @@ start_bcm()
...
-    elif [ "$onie_platform" == "x86_64-m3000-r1" ]; then
+    if [ $sonic_asic_type == "broadcom" ]; then
         start_bcm
-        DAEMON_ARGS+=" -p /etc/syncd.d/m3000_r1.profile"
-
-    elif [ "$onie_platform" == "x86_64-mlnx_x86-r5.0.1400" ] || [ "$onie_platform" == "x86_64-mlnx_msn2700-r0" ]; then
+        DAEMON_ARGS+=" -p $HWSKU_DIR/sai.profile "
+    elif [ $sonic_asic_type == "mellanox" ]; then
         start_mlnx
+        DAEMON_ARGS+=" -p /tmp/sai.profile "
```
Comment (stcheng) [inline debian/syncd.init]: question (may not be related): why mellanox sai.profile in /tmp/ folder instead of $HWSKU_DIR/ folder

Comment (lguohan) [inline debian/syncd.init]: see line 62 and 63. HWSKU_DIR is mounted as RO, so I cannot modify the file.

Comment (stcheng) [inline debian/syncd.init]: i see. thanks.

**Comment by stcheng** [comment]
Addressing https://github.com/Azure/sonic-sairedis/issues/167

**Pull request merged**